tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/AnnotationAttachmentTest.java

```diff
@@ -367,4 +369,89 @@ public void testAnnotsWithConstLists() {
         Assert.assertEquals(element.getKind(), NodeKind.LITERAL);
         Assert.assertEquals(((BLangLiteral) element).value, "test");
     }
+
```

Comment:
Sorry, missed this previously. But shall we add some runtime tests too? See org.ballerinalang.test.annotations.AnnotationRuntimeTest. The data provider at https://github.com/ballerina-platform/ballerina-lang/blob/master/tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/AnnotationRuntimeTest.java#L87 doesn't seem to be used though  (the test that used that seems to have gotten removed :/), so we'll have to add that back too.

Reply:
Basically let's test the following,
1. map type
2. record type with required fields that have defaults (at runtime we can check if the default was set)
3. array of maps or records

Reply:
Added runtime tests in b75e5a2. 
`testServiceAnnotAccess1` test is disabled since it was failing. Created https://github.com/ballerina-platform/ballerina-lang/issues/29033 to track it.